### ISSUE TITLE
[Feature Request]: DataGrid needs an accessible solution to overflowing values

### ISSUE THREAD

**Issue by davidmenendez** [carbon-design-system/ibm-products#5618]
### The problem

this issue is a follow up to #5351 

currently when using columns with resize it's possible to end up with a value that gets hidden with ellipses. that solution is not accessible and as a result to recent accessibility fixes to datagrid where unnecessary `title` attributes were being used as tooltips for overflowing values there's currently no way to view those values without resizing the column.

i haven't been able to find anything specifically regarding this accessibility issue when it comes to resizable tables, but according to [this article](https://yatil.net/blog/text-overflow-ellipsis-harmful) it's bad practice from an accessibility standpoint to hide values. so i think we have two options here

### The solution

1. set a minimum width to cells and add a horizontal scroll for values that extend beyond the cell
2. using `word-wrap: break-word` to truncate the value and adjust the row height to accommodate the value 

<img width="1056" alt="Screenshot 2024-06-27 at 9 02 06 AM" src="https://github.com/carbon-design-system/ibm-products/assets/6370760/64155459-4669-4cb5-9e96-668da29faf00">

the screenshot above shows an example of how a table width responsive row height would react to the cell truncation

### Owner/main maintainer(s)

David Menendez

### Second/backup maintainer(s)

Matt Gallo

### Business priority

High Priority = pressing release

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues
```[tasklist]
### Tasks
- [ ] Need to align with Data table in "core" Carbon
- [ ] Meet with Design maintainers
```

**Comment by davidmenendez** [comment]
@lakeside76

**Comment by davidmenendez** [comment]
@marinamas

**Comment by oliviaflory** [comment]
Ref: https://github.com/carbon-design-system/carbon/issues/5131

**Comment by elycheea** [comment]
REF# https://github.com/carbon-design-system/carbon/issues/5131

**Comment by ljcarot** [comment]
We've been investigating [TanStack ](https://tanstack.com/table/latest)[Table](https://tanstack.com/table/latest), a third-party, open-source offering, which provides extensive data table capabilities surpassing what our Carbon Datagrid offers. It provides much more flexibility and customization. TanStack Table is headless which means it can easily be added alongside Datagrid component in your product or application. The benefits of more flexibility for product teams and less maintenance for Carbon makes it a win win. Lastly, it is available in multiple frameworks including [React](https://github.com/carbon-design-system/tanstack-carbon/tree/main/react#readme) and [Web Component](https://github.com/carbon-design-system/tanstack-carbon/tree/main/web-components#readme) so it provides an option to non-React product teams.

For these reasons, we have decided to transition from building our own custom table component to using an example-based approach with TanStack Table. Datagrid will still exist in our library for existing teams but we are announcing the deprecation* of the Datagrid component in v2.54.0 release so teams can begin to work through the transition. Details about how to use both Datagrid and TanStack together can be found [here](https://github.com/carbon-design-system/tanstack-carbon/tree/main). 

*Deprecation means that no new features will be added however sev 1 and sev 2 bugs will be supported.
